test: fix healthcheck tests for Supabase connection

shifluxxc opened the pull request:
## Pull Request Template


**Description:**

-   Fixed tests for healthcheck functions related to Supabase SQL connections.
-   Improved logger/test setup so log assertions work as expected.
-   Addresses issues with failing tests and missing configuration for Supabase healthcheck logic.


**Breaking Changes (if applicable):**

-   No breaking changes introduced. All changes are backward compatible and only affect test logic and configuration handling.

**Additional Information:**

-   No new dependencies or external libraries were added.
-   Adjusted test utilities and logger setup to ensure logs are captured and asserted correctly.
-   Example: Now, tests pass even when `DB_HOST` or `DB_NAME` are omitted for Supabase, as they are set to default.

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

vikash commented on pkg/gofr/logging/mock_logger.go:
```diff
@@ -20,6 +20,9 @@ func NewMockLogger(level Level) Logger {
 	}
 }
 
+func (m *MockLogger) SetOut(w io.Writer) {
```
This is not required and also, should not be part of this PR.

coolwednesday commented:
@shifluxxc , Please ensure that the PR title clearly reflects the primary fix or feature you're introducing.
Also, we kindly request that you do not start working on an issue unless it has been assigned to you. This helps prevent duplication of effort and avoids situations where multiple contributors are unknowingly working on the same task in parallel.

Umang01-hash commented:
@shifluxxc Please comment on the issue that you want to work on it so that i can assign this issue to you. Thanks

shifluxxc commented:
> @shifluxxc , Please ensure that the PR title clearly reflects the primary fix or feature you're introducing. Also, we kindly request that you do not start working on an issue unless it has been assigned to you. This helps prevent duplication of effort and avoids situations where multiple contributors are unknowingly working on the same task in parallel.

I will make sure to wait for assignment before picking up issues . I updated the PR title and also removed the setOut function from mocklogger.

shifluxxc commented on pkg/gofr/datasource/sql/supabase_test.go:
```diff
@@ -122,12 +121,13 @@ func TestConfigureSupabaseConnection(t *testing.T) {
-				}
+				// if mockLogger, ok := mockLogger.(*logging.MockLogger); ok {
+				// 	mockLogger.SetOut(os.Stdout)
+				// }
```
I have removed the comment .

coolwednesday commented:
@shifluxxc , if you've addressed the review comments, please mark them as resolved. This helps move your PR toward merging. If the reviewer disagrees or has further suggestions, they will leave additional comments or update their review accordingly.

coolwednesday commented on go.work.sum:
```diff
@@ -844,6 +847,7 @@ github.com/go-gl/glfw/v3.3/glfw v0.0.0-20200222043503-6f7a984d4dc4 h1:WtGNWLvXpe
 github.com/go-gl/glfw/v3.3/glfw v0.0.0-20200222043503-6f7a984d4dc4/go.mod h1:tQ2UAYgL5IevRw8kRxooKSPJfGvJ9fJQFa0TUsXzTg8=
 github.com/go-jose/go-jose/v4 v4.0.4 h1:VsjPI33J0SB9vQM6PLmNjoHqMQNGPiZ0rHL7Ni7Q6/E=
 github.com/go-jose/go-jose/v4 v4.0.4/go.mod h1:NKb5HO1EZccyMpiZNbdUw/14tiXNyUJh188dfnMCAfc=
+github.com/go-jose/go-jose/v4 v4.0.5/go.mod h1:s3P1lRrkT8igV8D9OjyL4WRyHvjB6a4JSllnOrmmBOA=
```
Apologies for the iterative reviews, but it would be best to remove the changes in the go.work.sum file. Updates to this file are handled automatically during releases, and manual changes can lead to unnecessary conflicts.

shifluxxc replied:
successfully reverted the changes in go.work.sum file

coolwednesday commented:
I have tested the PR, it works fine. Below are the screenshots for the reference of other reviewers. However there is no logging and traces for HealthCheck in any of the SQL supported datasources addition(I checked for mysql local instance and supabase). I feel that is out of the scope of this PR and can be handled in another PR.

<img width="1152" alt="Screenshot 2025-06-17 at 1 18 09 PM" src="https://github.com/user-attachments/assets/e8561a99-e637-433a-972b-cca8d33a1a83" />

<img width="1177" alt="Screenshot 2025-06-17 at 1 18 20 PM" src="https://github.com/user-attachments/assets/ce9a7686-310a-4c51-9d5b-625b78cc8b18" />

shifluxxc commented:
> I have tested the PR, it works fine. Below are the screenshots for the reference of other reviewers. However there is no logging and traces for HealthCheck in any of the SQL supported datasources addition(I checked for mysql local instance and supabase). I feel that is out of the scope of this PR and can be handled in another PR.
> 
> @shifluxxc, please change the PR's name to : "Fixing HealthCheck for Supabase"
> 
> <img alt="Screenshot 2025-06-17 at 1 18 09 PM" width="1152" src="https://private-user-images.githubusercontent.com/98943137/455874434-e8561a99-e637-433a-972b-cca8d33a1a83.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNDgxMzAsIm5iZiI6MTc1MDE0NzgzMCwicGF0aCI6Ii85ODk0MzEzNy80NTU4NzQ0MzQtZTg1NjFhOTktZTYzNy00MzNhLTk3MmItY2NhOGQzM2ExYTgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE3VDA4MTAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxZTUwMTBjNzRlYjQxNzEzZDQzZjIyMzcwOTk0OGUyYzA1ZmQzZWI2ZDUzYmE1ZGMyNDg1N2M4YzIyZWNiYzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KocDSustoldDBahkGuEg6-APA9Alweijrtj9W4rKWCM"> <img alt="Screenshot 2025-06-17 at 1 18 20 PM" width="1177" src="https://private-user-images.githubusercontent.com/98943137/455874554-ce9a7686-310a-4c51-9d5b-625b78cc8b18.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNDgxMzAsIm5iZiI6MTc1MDE0NzgzMCwicGF0aCI6Ii85ODk0MzEzNy80NTU4NzQ1NTQtY2U5YTc2ODYtMzEwYS00YzUxLTlkNWItNjI1Yjc4Y2M4YjE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE3VDA4MTAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3YWMwZmUyM2IyNzI2M2Y0YTZmMTk1YzcwNjEyMTU2MDk4ZWE3NGI1YTM4YWU1MWNiNmFhZjA1YTZiM2Q5Y2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.N-QBiazpHs1vOTHLr3nm5F63zemwuiU0Iy9mT6LebSk">



> I have tested the PR, it works fine. Below are the screenshots for the reference of other reviewers. However there is no logging and traces for HealthCheck in any of the SQL supported datasources addition(I checked for mysql local instance and supabase). I feel that is out of the scope of this PR and can be handled in another PR.
> 
> @shifluxxc, please change the PR's name to : "Fixing HealthCheck for Supabase"
> 
> <img alt="Screenshot 2025-06-17 at 1 18 09 PM" width="1152" src="https://private-user-images.githubusercontent.com/98943137/455874434-e8561a99-e637-433a-972b-cca8d33a1a83.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNDgxOTYsIm5iZiI6MTc1MDE0Nzg5NiwicGF0aCI6Ii85ODk0MzEzNy80NTU4NzQ0MzQtZTg1NjFhOTktZTYzNy00MzNhLTk3MmItY2NhOGQzM2ExYTgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE3VDA4MTEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxMWU5ZTY4YjAzMzEwNjRkNzk1YmNkZDMyMThhYmI3YjIyZmU2YzU4ZGU3N2U0ZmRiZmY1NzAxODY5YmQ1NTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aTwyluU44Epp2L-SBRvY7l-xPaLw-3z4povoU4rQGdM"> <img alt="Screenshot 2025-06-17 at 1 18 20 PM" width="1177" src="https://private-user-images.githubusercontent.com/98943137/455874554-ce9a7686-310a-4c51-9d5b-625b78cc8b18.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNDgxOTYsIm5iZiI6MTc1MDE0Nzg5NiwicGF0aCI6Ii85ODk0MzEzNy80NTU4NzQ1NTQtY2U5YTc2ODYtMzEwYS00YzUxLTlkNWItNjI1Yjc4Y2M4YjE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE3VDA4MTEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0MTRkYTkzMWU4Y2Y4M2I1YjFmNzUxOTU2MzdlNGRkMjU5MDhkODlkOTdiYTU2M2E2NTRmODZiYTU4ODY1ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xOwP-QmAeQzP8k7cVCHiHVhT9JTJUteER4D_61JtCA8">

Done

Umang01-hash commented:
<img width="901" alt="Screenshot 2025-06-18 at 9 04 03 AM" src="https://github.com/user-attachments/assets/885e0ec0-bad7-4455-9893-0718a0c276f2" />

@shifluxxc You have the following linter issues in your PR. Please resolve them so that we can go ahead and re-review your PR.

shifluxxc commented:
> <img alt="Screenshot 2025-06-18 at 9 04 03 AM" width="901" src="https://private-user-images.githubusercontent.com/55635734/456302561-885e0ec0-bad7-4455-9893-0718a0c276f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAyMjU0OTcsIm5iZiI6MTc1MDIyNTE5NywicGF0aCI6Ii81NTYzNTczNC80NTYzMDI1NjEtODg1ZTBlYzAtYmFkNy00NDU1LTk4OTMtMDcxOGEwYzI3NmYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE4VDA1Mzk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0OWMxMTZiMTk0OWNlZDc4ZTAzOGJkZmUxNzUwODZjNDZjOThiYTZjMmVhNDE3OWM1MmY4NmRiNDhiYmRjOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WgpbQJ6vAN7025VeTBp3WBA-Zq7CjKUC6WyZdlOovcI">
> @shifluxxc You have the following linter issues in your PR. Please resolve them so that we can go ahead and re-review your PR.

Sir I have resolved the linting , you can review the changes
